tools/ReadConsoleInputStream/ConcurrentCircularQueue.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Nivot.Terminal
+{
+    /// <summary>
+    /// Implements a circular buffer.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class ConcurrentCircularQueue<T> : ConcurrentQueue<T>
+    {
+        public ConcurrentCircularQueue(int capacity)
+        {
+            Capacity = GetAlignedCapacity(capacity);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="collection"></param>
+        /// <param name="capacity"></param>
```

Comment:
We're all blank. Does this need to exist?

Reply:
Also going to rename this class to ConcurrentBoundedQueue due to aforementioned brainfart.